examples/DisplayTraits/Sample/CollectionViewController.m

```diff
@@ -0,0 +1,72 @@
+/* This file provided by Facebook is for non-commercial testing and evaluation
+ * purposes only.  Facebook reserves all rights not expressly granted.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * FACEBOOK BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#import "CollectionViewController.h"
+#import "KittenNode.h"
+
+@interface CollectionViewController () <ASCollectionDelegate, ASCollectionDataSource>
+@property (nonatomic, strong) ASCollectionNode *collectionNode;
+@end
+
+@implementation CollectionViewController
+
+- (instancetype)init
+{
+  UICollectionViewFlowLayout *layout = [[UICollectionViewFlowLayout alloc] init];
+  layout.minimumLineSpacing = 10;
+  layout.minimumInteritemSpacing = 10;
+  
+  ASCollectionNode *collectionNode = [[ASCollectionNode alloc] initWithCollectionViewLayout:layout];
+  
+  if (!(self = [super initWithNode:collectionNode]))
+    return nil;
+  
+  self.title = @"Collection Node";
+  _collectionNode = collectionNode;
+  collectionNode.dataSource = self;
+  collectionNode.delegate = self;
+  return self;
+}
+
+- (void)viewDidLoad
+{
+  [super viewDidLoad];
+  self.collectionNode.view.contentInset = UIEdgeInsetsMake(20, 10, CGRectGetHeight(self.tabBarController.tabBar.frame), 10);
+}
+
+#pragma mark - ASCollectionDataSource
+
+- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section
+{
+  return 50;
+}
+
+- (ASCellNode *)collectionView:(ASCollectionView *)collectionView nodeForItemAtIndexPath:(NSIndexPath *)indexPath
+{
+  KittenNode *cell = [[KittenNode alloc] init];
+  cell.textNode.maximumNumberOfLines = 3;
+  cell.imageTappedBlock = ^{
+    [KittenNode defaultImageTappedAction:self];
+  };
+  return cell;
+}
+
+- (ASSizeRange)collectionView:(ASCollectionView *)collectionView constrainedSizeForNodeAtIndexPath:(NSIndexPath *)indexPath
+{
+  ASDisplayTraits displayTraits = self.collectionNode.environmentState.displayTraits;
```

Comment:
Maybe we should provide a better API to get to the displayTraits so the consumer does not need to have to know (and I think should not know) about the environmentState

cc @appleguy